File: apps/launcher/ui/api/sync_client.py
from __future__ import annotations

from apps.launcher.ui.api.base_client import BaseClient, ApiResult


class SyncClient(BaseClient):
    def snapshot(self) -> ApiResult: return self.get("/api/v1/sync/snapshot")
    def changes(self, cursor: str | None = None) -> ApiResult:
        suffix = f"?cursor={cursor}" if cursor else ""
        return self.get(f"/api/v1/sync/changes{suffix}")

File: apps/mobile/android/app/src/main/java/br/com/aipinho/mobile/network/SyncClient.kt
package br.com.aipinho.mobile.network
import br.com.aipinho.mobile.models.ConnectionProfile
class SyncClient(profile:ConnectionProfile,token:()->String?):BaseApiClient(profile,profile.corePort,token) {
    fun snapshot()=get("/api/v1/sync/snapshot")
    fun changes(cursor:String)=get("/api/v1/sync/changes?cursor=$cursor")
}
